src/main/java/com/diffplug/gradle/spotless/FormatTask.java

```diff
@@ -53,32 +70,41 @@ public String getFormatName() {
 	}
 
 	@TaskAction
-	public void format() throws Exception {
+	public void format(IncrementalTaskInputs inputs) throws Exception {
 		if (target == null) {
 			throw new GradleException("You must specify 'Iterable<File> toFormat'");
 		}
 		// combine them into the master formatter
-		Formatter formatter = new Formatter(lineEndingsPolicy, encoding, getProject().getProjectDir().toPath(), steps);
+		Formatter formatter = Formatter.builder()
+				.lineEndingPolicy(lineEndingPolicy)
+				.encoding(encoding)
+				.projectDirectory(getProject().getProjectDir().toPath())
+				.steps(steps)
+				.build();
 
 		// perform the check
 		if (check) {
-			formatCheck(formatter);
+			formatCheck(formatter, inputs);
 		} else {
-			formatApply(formatter);
+			formatApply(formatter, inputs);
 		}
 	}
 
 	/** Checks the format. */
-	private void formatCheck(Formatter formatter) throws IOException {
+	private void formatCheck(Formatter formatter, IncrementalTaskInputs inputs) throws IOException {
 		List<File> problemFiles = new ArrayList<>();
 
-		for (File file : target) {
+		inputs.outOfDate(input -> {
+			File file = input.getFile();
 			getLogger().debug("Checking format on " + file);
 			// keep track of the problem toFormat
-			if (!formatter.isClean(file)) {
-				problemFiles.add(file);
-			}
-		}
+			Errors.rethrow()
```

Comment:
It's to do with how, in Java 8, lambdas cannot throw checked exceptions unless the interface they target declares that it `throws Throwable`. `IncrementalTaskInputs.outOfDate(Action)` accepts a Groovy `Action`, which doesn't `throw Throwable`, so all checked exceptions thrown within the Action must be wrapped in unchecked exceptions. Therefore, because `!formatter.isClean(file)` throws `IOException` - a checked exception - it has to be wrapped in an unchecked exception of sorts, otherwise it won't compile.

In this case, I decided to use Durian's `Errors.rethrow()` mechanism to wrap all `IOException`s as `Errors.WrappedAsRuntimeException`s, although I could have just as easily wrapped them in `UncheckedIOException`s instead (but that would arguably be less readable, as it would have required `try { ... } catch(IOException e) { throw new UncheckedIOException(e); }`).